File: prerendertron.js
#!/usr/bin/env node

const http = require('http')
const https = require('https')

const hitCache = slug => {
  const req = http.request({
    hostname: 'localhost',
    port: 3000,
    path: `/render/https://beautifultrouble.org/toolbox/#/tool/${slug}`,
    method: 'GET'
  }, res => {
    res.on('end', () => process.stdout.write(`loaded\n`))
  })
}

const req = https.request({
  hostname: 'api.beautifulrising.org',
  port: 443,
  path: '/api/v1/modules',
  method: 'GET'
}, res => {
  let body = ""

  res.on('data', chunk => {
    body += chunk
  })
  res.on('end', () => {
    const tools = JSON.parse(body);
    tools.forEach(tool => {
      process.stdout.write(`Loading ${tool.slug}... `)
      hitCache(tool.slug)
    })
  })
})

req.end()

File: prerendertron.py
#!/usr/bin/env python

import requests

RENDERTRON_PORT = 3000


j = requests.get("https://api.beautifultrouble.org/v2/en/toolbox.json").json()

for slug, tool in j["tools"].items():
    for lang in tool["langs"]:
        print(f"Rendering {tool['title']} ({lang})")
        requests.get(
            f"http://127.0.0.1:{RENDERTRON_PORT}/render/"
            + "https://beautifultrouble.org/toolbox/"
            + ("" if lang == "en" else f"{lang}/")
            + f"{slug}?refreshCache=true"
        )
